app/test/frontend/static_files_test.dart

```diff
@@ -77,5 +77,35 @@ void main() {
         expect(f.etag.contains('mocked_hash_'), isFalse);
       });
     }
+
+    test('proper hash in css content', () {
+      final cssHashes = <String, String>{};
+      final css = cache.getFile('/static/css/style.css');
+      for (Match m in RegExp('url\\("(.*?)"\\);')
+          .allMatches(css.contentAsString)) {
+        final matched = m.group(1);
+        if (matched.contains('data:image')) continue;
+        final uri = Uri.parse(matched);
+        final absPath = p.normalize(p.join('/static/css', uri.path));
+        final hash = uri.queryParameters['hash'] ?? '_no_hash_';
+        if (cssHashes.containsKey(absPath) && cssHashes[absPath] != hash) {
+          throw Exception(
+              'Multiple hash for key: $absPath ($hash vs ${cssHashes[absPath]})');
+        }
+        cssHashes[absPath] = hash;
+      }
+
+      final expectedHashes = <String, String>{};
+      final containedFiles = [
```

Comment:
This is a really tightly coupled test... instead of collecting `cssHashes` and comparing that...
How about we just do:
```dart
if (hash != cache.getFile(absPath).etag) {
  throw ....
}
```
Inside the loop above... that way we don't have to touch this test every time we tweak the CSS